Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+    if args.get('description') is not None:
+        kwargs.update({'Description': args.get('description')})
+    if args.get('encrypted') is not None:
+        kwargs.update({'Encrypted': True if args.get('ebsEncrypted') == 'True' else False})
+    if args.get('kmsKeyId') is not None:
+        kwargs.update({'KmsKeyId': args.get('kmsKeyId')})
+
+    response = client.copy_image(**kwargs)
+    data = ({
+        'ImageId': response['ImageId'],
+        'Region': obj['_user_provided_options']['region_name']
+    })
+
+    ec = {'AWS.EC2.Images(val.ImageId === obj.ImageId)': data}
+    human_readable = tableToMarkdown('AWS EC2 Images', data)
+    return_outputs(human_readable, ec)
+
+
+def copy_snapshot_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    obj = vars(client._client_config)
+    kwargs = {
+        'SourceSnapshotId': args.get('sourceSnapshotId'),
+        'SourceRegion': args.get('sourceRegion'),
+    }
+    if args.get('description') is not None:
+        kwargs.update({'Description': args.get('description')})
+    if args.get('encrypted') is not None:
+        kwargs.update({'Encrypted': True if args.get('ebsEncrypted') == 'True' else False})
+    if args.get('kmsKeyId') is not None:
+        kwargs.update({'KmsKeyId': args.get('kmsKeyId')})
+
+    response = client.copy_snapshot(**kwargs)
+    data = ({
+        'SnapshotId': response['SnapshotId'],
+        'Region': obj['_user_provided_options']['region_name']
+    })
+
+    ec = {'AWS.EC2.Snapshots(val.SnapshotId === obj.SnapshotId)': data}
+    human_readable = tableToMarkdown('AWS EC2 Snapshots', data)
+    return_outputs(human_readable, ec)
+
+
+def describe_reserved_instances_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+    output = []
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('offeringClass') is not None:
+        kwargs.update({'OfferingClass': args.get('offeringClass')})
+    if args.get('reservedInstancesIds') is not None:
+        kwargs.update({'ReservedInstancesIds': parse_resource_ids(args.get('reservedInstancesIds'))})
+
+    response = client.describe_reserved_instances(**kwargs)
+    for i, reservation in enumerate(response['ReservedInstances']):
+        data.append({
+            'ReservedInstancesId': reservation['ReservedInstancesId'],
+            'Start': datetime.strftime(reservation['Start'], '%Y-%m-%dT%H:%M:%SZ'),
+            'End': datetime.strftime(reservation['End'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.